crates/ibc/src/core/ics04_channel/error.rs

```diff
@@ -19,6 +19,8 @@ pub enum ChannelError {
     Connection(connection_error::ConnectionError),
     /// port error: `{0}`
     Port(port_error::PortError),
+    /// port `{port_id}` is unknown
+    PortNoSource { port_id: PortId },
```

Comment:
yes good point; I wanted to avoid using `PortError` because I think we should get rid of it, but this is better for now

[c7f10ad](https://github.com/cosmos/ibc-rs/pull/347/commits/c7f10adfa0024037669515b85f5936c4997fb425)